github-runner-manager/tests/unit/test_pre_job_script.py

```diff
@@ -465,6 +468,95 @@ def test_allow_external_contributor_enabled_skips_check(
     assert LOG_CHECK_PASSED not in result.stderr
 
 
+@pytest.mark.parametrize(
+    "author_association,should_pass",
+    [
+        ("OWNER", True),
+        ("MEMBER", True),
+        ("COLLABORATOR", True),
+        ("CONTRIBUTOR", True),  # Allowed for private repos
+        ("FIRST_TIME_CONTRIBUTOR", False),  # Still blocked
+        ("NONE", False),  # Still blocked
+    ],
+)
+def test_private_repository_allows_contributor(
+    pre_job_template: Template,
+    github_env_vars: Dict[str, str],
+    tmp_path: Path,
+    author_association: str,
+    should_pass: bool,
+    default_template_vars: Dict,
+):
+    """Test that private repositories extend allowed authorizations to include CONTRIBUTOR.
+
+    Private repositories trust CONTRIBUTORs since they have repository access,
+    but still block FIRST_TIME_CONTRIBUTOR and NONE.
+    """
+    # Create a fork PR event for a private repository
+    private_repo_event = _create_github_event_payload(
+        author_association=author_association,
+        is_fork=True,
+        is_private=True,
+    )
+
+    result = render_and_execute_script(
+        template=pre_job_template,
+        template_vars=default_template_vars,
+        env_vars=github_env_vars,
+        github_event=private_repo_event,
+        tmp_path=tmp_path,
+    )
+
+    if should_pass:
+        assert result.returncode == 0, (
+            f"Expected exit code 0 for private repository with {author_association}, "
+            f"got {result.returncode}\nstderr: {result.stderr}"
+        )
+        assert "Private repository - extended authorization includes CONTRIBUTOR" in result.stderr
+        assert LOG_AUTH_FAILED not in result.stderr
+    else:
+        assert result.returncode == 1, (
+            f"Expected exit code 1 for private repository with {author_association}, "
+            f"got {result.returncode}\nstderr: {result.stderr}"
+        )
+        assert LOG_AUTH_FAILED in result.stderr
+
+
+def test_public_repository_blocks_contributor(
+    pre_job_template: Template,
+    github_env_vars: Dict[str, str],
+    tmp_path: Path,
+    default_template_vars: Dict,
+):
+    """Test that public repositories still block CONTRIBUTOR role.
+
+    This ensures the behavior change only applies to private repositories.
+    """
```

Comment:
Can you change the doc string to the "arrange act assert" pattern we usually use?